src/hammer-vlsi/hammer_vlsi/hammer_tool.py

```diff
@@ -914,6 +914,49 @@ def get_bumps(self) -> Optional[BumpsDefinition]:
             pitch=self.get_setting("vlsi.inputs.bumps.pitch"),
             cell=self.get_setting("vlsi.inputs.bumps.cell"), assignments=assignments)
 
+    def get_pin_assignments(self) -> List[PinAssignment]:
+        """
+        Get a list of pin assignments in accordance with settings in the Hammer IR.
+        :return: A potentially empty list of PinAssigments.
+        """
+        pin_mode = str(self.get_setting("vlsi.inputs.pin_mode"))  # type: str
+        if pin_mode == "none":
+            return []
+        elif pin_mode != "generated":
+            self.logger.error(
+                "Invalid pin_mode {mode}. Using none pin mode.".format(mode=pin_mode))
+            return []
+
+        # Generated pin mode needs to ingest the assignments
+        assigns = []  # type: List[PinAssignment]
+        for raw_assign in self.get_setting("vlsi.inputs.pin.assignments"):
+            pins = str(raw_assign["pins"])  # type: str
+            side = None if not "side" in raw_assign else raw_assign["side"]
+            if not (side == None or side == "top" or side == "bottom" or side == "right" or side == "left") :
```

Comment:
Do we want an enum for side? I'm leaning no, but worth bringing up.

Reply:
It seemed excessive to me but I don't much care either way.